util/network-alert/src/notifier.rs

```diff
@@ -44,23 +44,26 @@ impl Notifier {
         self.received_alerts.insert(alert.id, Arc::clone(&alert));
 
         // check conditions, figure out do we need to notice this alert
-        if alert
-            .min_version
-            .as_ref()
-            .map(|min_v| Version::parse(&self.client_version) < Version::parse(min_v))
-            == Some(true)
-        {
-            return;
-        }
+        if let Ok(client_version) = Version::parse(&self.client_version) {
+            if let Some(ref min_v) = alert.min_version {
+                if Version::parse(&min_v)
+                    .map(|min_v| client_version < min_v)
+                    .unwrap_or(true)
+                {
+                    return;
+                }
+            }
 
-        if alert
-            .max_version
-            .as_ref()
-            .map(|max_v| Version::parse(&self.client_version) > Version::parse(max_v))
-            == Some(true)
-        {
-            return;
+            if let Some(ref max_v) = alert.max_version {
+                if Version::parse(&max_v)
+                    .map(|max_v| client_version > max_v)
+                    .unwrap_or(true)
+                {
+                    return;
+                }
+            }
```

Comment:
It's better to define a helper function, like `is_allowed_version` or else.

Reply:
Please add unit tests.

Reply:
the old test case is works well https://github.com/nervosnetwork/ckb/blob/develop/util/network-alert/src/tests/test_notifier.rs#L6

this PR just explicit the program's behaviour